docs/development/quickstart.rst

```diff
@@ -0,0 +1,335 @@
+  process outlined in the :ref:`testing-guidelines` section.
+- **Build documentation**: If you are updating the documentation, you will want to
+  :ref:`build the documentation <builddocs>` to ensure that it looks good.
+- **Add a changelog entry**: For most code changes you will need to
+  :ref:`add-changelog`.
+
+.. tip:: For more information and examples see :ref:`edit-flow` section.
+
+You can see a summary of the changes you've currently made by running:
+
+.. code-block:: shell
+
+    git status
+
+You can then commit your all your changes to your local repository with an explanatory
+`commit message <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_:
+
+.. code-block:: shell
+
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+.. Important:: Never merge changes from ``upstream/main`` into your feature branch. If
+   changes in ``main`` require changes to our code you must :ref:`rebase`.
+
+.. _contributing.push-code:
+
+Pushing your changes
+--------------------
+
+When you want your changes to appear publicly on your GitHub page, push your
+forked feature branch's commits::
+
+    git push origin --set-upstream subpackage-bug-fix
+
+Here ``origin`` is the default name given to your fork on GitHub.
+
+Now your code is on GitHub, but it is not visible to the Astropy maintainers. For that
+to happen, a pull request needs to be submitted on GitHub.
+
+The first time you push to a new branch on GitHub, you will see a message like below
+with a useful link to create a pull request::
+
+  remote: Create a pull request for 'subpackage-bug-fix' on GitHub by visiting:
+  remote:      https://github.com/YOUR-USER-NAME/astropy/pull/new/subpackage-bug-fix
+
+
+.. _quickstart-pull-request:
+
+Making a pull request
+---------------------
+
+If everything looks good, you are ready to make a pull request (PR). A PR is how
+code from your local repository becomes available to the GitHub community to review and
+merged into project to appear the in the next release.
+
+Most of the time you can just follow the link that ``git`` provided when you pushed
+your branch and create the PR. If you don't have that link (and for a few more details), you can follow the :ref:`pull-request` instructions.
+
+Follow the instructions in the PR template and fill it out as completely as possible.
+
+If your PR is still a work in progress then instead of clicking "Create pull request",
+click on the small down arrow next to it and select "`Create draft pull request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests>`__".
+In addition, if your commits are not ready for CI testing, you
+should include ``[ci skip]`` the last commit message (but note that code formatting
+checks and documentation building will still be done).
```

Comment:
```suggestion
should include ``[ci skip]`` the last commit message – but note that code formatting
checks and documentation building will still be done. Formatting and style error *should*
already have been fixed before committing if you did :ref:`_contributing.pre-commit`,
but if you have not, you can use the :ref:`pre-commit_bot` to fix them automatically in the PR.
```
Sorry, my RST skills are very lame; I just hope that the first reference points to the quick intro inside this file.

Reply:
I don't think you should include the leading underscore.

```rst
:ref:`contributing.pre-commit`
```